src/Model/ChangeStreamIterator.php

```diff
@@ -47,7 +49,7 @@
  *
  * @internal
  * @template TValue of array|object
- * @template-extends IteratorIterator<int, TValue, Cursor<TValue>>
+ * @template-extends IteratorIterator<int, TValue, Iterator<int, TValue>>
```

Comment:
After checking, psalm seems to accept union types here.
```suggestion
 * @template-extends IteratorIterator<int, TValue, CursorInterface&Iterator<int, TValue>>
```